pkg/spireentry/reconciler.go

```diff
@@ -250,9 +258,46 @@ func (r *entryReconciler) recalculateUnsupportFields(ctx context.Context, log lo
 	r.nextGetUnsupportedFields = time.Now().Add(10 * time.Minute)
 }
 
-func (r *entryReconciler) listEntries(ctx context.Context) ([]spireapi.Entry, error) {
+func (r *entryReconciler) shouldProcessOrDeleteEntryID(entry spireapi.Entry) (bool, bool) {
+	if r.config.EntryIDPrefix == "" {
+		return true, false
+	}
+	if strings.HasPrefix(entry.ID, r.config.EntryIDPrefix) {
+		return true, false
+	}
+	if r.config.EntryIDPrefixCleanup != nil {
+		cleanupPrefix := *r.config.EntryIDPrefixCleanup
+		if cleanupPrefix == "" {
+			if strings.Contains(entry.ID, "/") {
+				return false, false
+			}
+			return false, true
```

Comment:
Dot is the separator between prefix, correct? Also this can be simplified.
```suggestion
			return false, strings.Contains(entry.ID, ".") {
```

Reply:
Close... it needs to be negated though... Will rework it that way though to be this pattern.

Reply:
Yep, thanks for catching that.